pandas/tests/libs/test_hashtable.py

```diff
@@ -0,0 +1,238 @@
+import numpy as np
+import pytest
+
+from pandas._libs import hashtable as ht
+
+import pandas._testing as tm
+
+
+@pytest.mark.parametrize(
+    "table_type, dtype",
+    [
+        (ht.Int64HashTable, np.int64),
+        (ht.UInt64HashTable, np.uint64),
+        (ht.Float64HashTable, np.float64),
+        (ht.Int32HashTable, np.int32),
+        (ht.UInt32HashTable, np.uint32),
+    ],
+)
+class TestHashTable:
+    def test_get_set_contains_len(self, table_type, dtype):
+        index = 5
+        table = table_type(55)
+        assert len(table) == 0
+        assert index not in table
+        table.set_item(index, 42)
+        assert len(table) == 1
+        assert index in table
+        assert table.get_item(index) == 42
+        table.set_item(index + 1, 41)
+        assert index in table
+        assert index + 1 in table
+        assert len(table) == 2
+        table.get_item(index) == 42
+        table.get_item(index + 1) == 41
+        table.set_item(index, 21)
+        assert index in table
+        assert index + 1 in table
+        assert len(table) == 2
+        table.get_item(index) == 21
+        table.get_item(index + 1) == 41
+        assert index + 2 not in table
+        with pytest.raises(KeyError) as excinfo:
+            table.get_item(index + 2)
+        assert str(index + 2) in str(excinfo.value)
```

Comment:
theres a lot going on in this test.  can it either be split up into smaller pieces, or at least have some whitespace clarify what assertions are connected to what calls

Reply:
thanks